enhancements/cluster-logging/elasticsearch-hot-warm-architecture.md

```diff
@@ -0,0 +1,156 @@
+
+## Motivation
+
+Currently our managed Elasticsearch cluster is restricted in the amount of days of retention we can provide in part due to the amount of metadata ES keeps in memory related to its indices. If we are taking advantage of the different index management APIs provided by Elasticsearch we can seek to better tune clusters to increase retention and overall performance.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the addition to the API to provide a means to configure WARM policies
+
+* Replace|repurpose the current rollover cronjob to also perform hot to warm transition actions
+
+We will be successful when:
+
+* Users are able to configure WARM policies so that we can optimize their cluster usage
+
+* We are able to transition a user's index from one policy to the next based on their configuration
+
+### Non-Goals
+
+* We will not be marking different nodes for the different policies and reallocating shards based on that
+
+* We will not be implementing this using the Opendistro plugin
+
+## Proposal
+
+Currently we offer a mechanism for users to configure their index lifecycles based on two states: hot and delete. By default indices are created and stay in the hot state, where they can be rolled over based on a configuration, however they stay in a "hot" state until they are eventually deleted based on the delete policy.
+
+Elasticsearch offers mechanisms to move indices between hot, warm, cold, and delete and discusses approaches for this [in a blog article](https://www.elastic.co/blog/implementing-hot-warm-cold-in-elasticsearch-with-index-lifecycle-management). However, this functionality is a x-pack feature and we do not support these. [Opendistro](https://opendistro.github.io/for-elasticsearch-docs/docs/ism/) offers a way to do this as well, however their plugins require ES7. We achieve our current functionality using cronjobs -- one for delete and one for rollover.
+
+Within the context of Elasticsearch ILM:
+1) Moving the index into a Warm state means the index is force merged to use less primary shards, its replica count is reduced, and the index priority is also reduced. The idea of this being that these indices are not being written to, may not be queried against as much so they do not need to be as distributed, and can be a lower priority when it comes to index recovery (meaning they would be recovered after high priority indices are).
+
+1) Moving into a Cold state further reduces its index priority to be lower than Warm (but still higher than delete) and, if one was using specific nodes to house indices of certain state it would move from Warm to Cold nodes. This is different than Closing an index because the index can still be queried (if one were to Close an index it would need to be Opened first before it can be searched).
```

Comment:
Same here, how does cold state translate in a number of primary shards and/or replica count?

Reply:
As this block states, in ILM indices move from warm to cold and the only change from one state to the other is the index priority.